File: problems/algoexpert/recursion/power_set.py
def main() -> None:
    sum = 0
    for s in powerset([n for n in range(1, 27)]):
        for v in s:
            sum += v
    print(sum)


def powerset(nums: list[int]) -> list[list[int]]:
    subsets: list[list[int]] = [[]]
    index = 0
    while index < 2**len(nums):
        subset: list[int] = []
        for i, num in enumerate(nums):
            if index & (1 << i) > 0:
                subset.append(num)
        subsets.append(subset)
        index += 1
    return subsets


if __name__ == "__main__":
    main()

File: problems/algoexpert/recursion/power_set.go
package main

import (
	"fmt"
	"math"
)

func main() {
	size := 26
	tt := make([]int, size)
	for i := 1; i <= size; i++ {
		tt[i-1] = i
	}
	sum := 0
	for _, s := range PowerSet(tt) {
		for _, v := range s {
			sum += v
		}
	}
	fmt.Println(sum)
}

func PowerSet(nums []int) [][]int {
	powerSetSize := int(math.Pow(2, float64(len(nums))))
	subsets := make([][]int, 0, powerSetSize)
	var index int
	for index < powerSetSize {
		var subSet []int
		for i, num := range nums {
			if index&(1<<uint(i)) > 0 {
				subSet = append(subSet, num)
			}
		}
		subsets = append(subsets, subSet)
		index++
	}
	return subsets
}

